src/hotspot/share/services/threadService.hpp

```diff
@@ -103,8 +103,8 @@ class ThreadService : public AllStatic {
 
   static jlong get_total_thread_count()       { return _total_threads_count->get_value(); }
   static jlong get_peak_thread_count()        { return _peak_threads_count->get_value(); }
-  static jlong get_live_thread_count()        { return _atomic_threads_count; }
-  static jlong get_daemon_thread_count()      { return _atomic_daemon_threads_count; }
+  static int get_live_thread_count()          { return _atomic_threads_count; }
+  static int get_daemon_thread_count()        { return _atomic_daemon_threads_count; }
```

Comment:
I don't follow. The fields are int so cast them to jlong before returning them. All the callers of these methods expect jlong so there can't be any issue there.

Reply:
You don't need to cast from int to jlong.   The callers of these methods expect int:

```
src/hotspot/share/services/threadService.cpp:1103:55: warning: conversion from 'jlong' {aka 'long int'} to 'int' may change value [-Wconversion]
 1103 |   int init_size = ThreadService::get_live_thread_count();
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
```

Changing int to jlong here, gets a new warning:

```
src/hotspot/share/services/threadService.cpp:1104:54: warning: conversion from 'jlong' {aka 'long int'} to 'int' may change value [-Wconversion]
 1104 |   _threads_array = new GrowableArray<instanceHandle>(init_size);
      |                                                      ^~~~~~~~~
```

If you look up in the header file, the field is an int.  It makes sense to return an int:

```
  static volatile int  _atomic_threads_count;
  static volatile int  _atomic_daemon_threads_count;
```

Correcting the types means returning a type that matches the declaration so that its consistent.

Reply:
I'd missed the GrowableArray code - the other callers expect jlong/643-bit.